test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ComplexTypeInheritance/ComplexTypeInheritanceTests.cs

```diff
@@ -539,6 +585,62 @@ public async Task PutCollectionComplexTypeProperty(string modelMode)
                 contentOfString));
         }
 
+        [Theory]
+        [MemberData(nameof(PostToCollectionNewComplextTypeMembers))]
+        // POST ~/Windows(3)/OptionalShapes
+        public async Task PostToCollectionComplexTypeProperty(string modelMode, string jObject, string targetPropertyResource, bool returnRepresentation)
+        {
+            //Arrange
+            string serviceRootUri = string.Format("{0}/{1}", BaseAddress, modelMode).ToLower();
+            string requestUri = serviceRootUri + "/Windows(3)/"+ targetPropertyResource;
+
+            //send a get request to get the current count
+            int count = 0;
+            using (HttpResponseMessage getResponse = await this.Client.GetAsync(requestUri))
+            {
+                getResponse.EnsureSuccessStatusCode();
+
+                var json = await getResponse.Content.ReadAsObject<JObject>();
+                var state = json.GetValue("value") as JArray;
+                count = state.Count;
+            }
+
+            //Setup the post request
+            var requestForPost = new HttpRequestMessage(HttpMethod.Post, requestUri);
+            requestForPost.Content = new StringContent(content:jObject, encoding: Encoding.UTF8, mediaType: "application/json");
+            if (returnRepresentation)
+            {
+                requestForPost.Headers.Add("Prefer", "return=representation");
+            }
+
+            //Act & Assert
+            HttpResponseMessage response = await Client.SendAsync(requestForPost);
+            string contentOfString = await response.Content.ReadAsStringAsync();
+
+            if(returnRepresentation)
+            {
+                JObject contentOfJObject = await response.Content.ReadAsObject<JObject>();
+                var result = contentOfJObject.GetValue("value") as JArray;
```

Comment:
>result [](start = 20, length = 6)

It looks only the count needed, you can use "$count" segment to get the count directly.

Reply:
Yes, he could make a separate GET request to /$count, but here he's specifically checking the result of the POST request.

---
In reply to: [223519003](https://github.com/OData/WebApi/pull/1628#discussion_r223519003) [](ancestors = 223519003)

Reply:
I agree that I could make the first request to get the count. With that being said, I do not see a major difference for my purpose and I felt doing a get was simpler.